www/data_service/src/classes/base.service.coffee

```diff
@@ -0,0 +1,47 @@
+class Base extends Factory
+    constructor: ($log, dataService, socketService, helperService) ->
+        return class BaseInstance
+            constructor: (o, @endpoint) ->
+                if not angular.isString(@endpoint)
+                    throw "#{endpoint} is not a string"
+
+                @update(o)
+
+                # get the id of the class type
+                classId = helperService.classId(@endpoint)
+                @id = @[classId]
+
+                # subscribe for WebSocket events
+                listener = (data) =>
+                    key = data.k
+                    message = data.m
+                    # filter for relevant messag
+                    @streamRegex = ///^#{@endpoint}\/#{@id}\/\w+$///g
+                    if @streamRegex.test(key)
+                        @update(message)
+                @unsubscribe = socketService.eventStream.subscribe(listener)
+                @listenerId = listener.id
+
+            update: (o) ->
+                angular.extend(@, o)
+
+            get: (args...) ->
+                dataService.get(@endpoint, @id, args...)
+
+            done: ->
+                # unsubscribe childs
+                for k, v of @
```

Comment:
I'd rather use `this` instead of `@`. It is a little strange to put just a symbol here so I think more verbose might be better :)